addons/l10n_sa_edi/data/sa_zatca_template.xml

```diff
@@ -0,0 +1,333 @@
+                                                                <ds:X509IssuerName xmlns:ds="http://www.w3.org/2000/09/xmldsig#" id="issuer_name" />
+                                                                <ds:X509SerialNumber xmlns:ds="http://www.w3.org/2000/09/xmldsig#" id="serial_number" />
+                                                            </xades:IssuerSerial>
+                                                        </xades:Cert>
+                                                    </xades:SigningCertificate>
+                                                </xades:SignedSignatureProperties>
+                                            </xades:SignedProperties>
+                                        </xades:QualifyingProperties>
+                                    </ds:Object>
+                                </ds:Signature>
+                            </sac:SignatureInformation>
+                        </sig:UBLDocumentSignatures>
+                    </ext:ExtensionContent>
+                </ext:UBLExtension>
+            </ext:UBLExtensions>
+        </template>
+
+        <template id="export_sa_zatca_ubl_signed_properties">
+            <xades:SignedProperties xmlns:xades="http://uri.etsi.org/01903/v1.3.2#" Id="xadesSignedProperties">
+                <xades:SignedSignatureProperties>
+                    <xades:SigningTime t-out="signing_time" />
+                    <xades:SigningCertificate>
+                        <xades:Cert>
+                            <xades:CertDigest>
+                                <ds:DigestMethod xmlns:ds="http://www.w3.org/2000/09/xmldsig#" Algorithm="http://www.w3.org/2001/04/xmlenc#sha256" />
+                                <ds:DigestValue xmlns:ds="http://www.w3.org/2000/09/xmldsig#" t-out="public_key_hashing" />
+                            </xades:CertDigest>
+                            <xades:IssuerSerial>
+                                <ds:X509IssuerName xmlns:ds="http://www.w3.org/2000/09/xmldsig#" t-out="issuer_name" />
+                                <ds:X509SerialNumber xmlns:ds="http://www.w3.org/2000/09/xmldsig#" t-out="serial_number" />
+                            </xades:IssuerSerial>
+                        </xades:Cert>
+                    </xades:SigningCertificate>
+                </xades:SignedSignatureProperties>
+            </xades:SignedProperties>
+        </template>
+
+        <template id="export_sa_zatca_invoice_line" inherit_id="account_edi_ubl.export_ubl_invoice_line" primary="True">
+            <xpath expr="//*[local-name()='TaxAmount']" position="replace">
+                <cbc:TaxAmount xmlns="urn:oasis:names:specification:ubl:schema:xsd:Invoice-2"
+                    xmlns:cbc="urn:oasis:names:specification:ubl:schema:xsd:CommonBasicComponents-2"
+                    t-att-currencyID="line.currency_id.name"
+                    t-esc="format_monetary(abs(tax_detail_vals['tax_amount_currency']))" />
```

Comment:
The value printed in the TaxAmount element also changes to the absolute amount of the tax. Instead of having two xpaths, one to change the attribute, and another to insert an element right after, is it not more efficient to replace ?

Reply:
@mebe-odoo I did not notice the abs and I don't think that in the tax_details it is often negative (or even in the original UBL)

Reply:
Fixed